datamodel/equal.go

```diff
@@ -25,6 +25,9 @@ package datamodel
 // so an error should only happen if a Node implementation breaks that contract.
 // It is generally not recommended to call DeepEqual on ADL nodes.
 func DeepEqual(x, y Node) bool {
+	if x == nil || y == nil {
```

Comment:
you don't want to `true` if _both_ are `nil`?

Reply:
I guess this goes into whether an untyped `nil` node is valid. I think `DeepEqual(nil, nil)` should indeed return true, and that's consistent with `reflect.DeepEqual` and what most Go users would expect.

You don't even need another `if` branch for this; you can just do:

```
if x == nil || y == nil {
    return x == y
}
```

Reply:
Also, I did write table-driven tests for DeepEqual, so please add coverage for this :)